### ISSUE TITLE
python3Packages.pelican tests fail

### ISSUE THREAD

**Issue by Moredread** [NixOS/nixpkgs#83513]
**Describe the bug**

Building `pelican` fails due to two broken tests on `master` and the `20.03` channel.

**To Reproduce**

```
> nix run -f channel:nixos-unstable python3Packages.pelican -L
(...)
ython3.7-pelican> test_custom_pagination_pattern (pelican.tests.test_paginator.TestPage) ... ok
python3.7-pelican> test_save_as_preservation (pelican.tests.test_paginator.TestPage) ... ok
python3.7-pelican> test_basic_generation_works (pelican.tests.test_pelican.TestPelican) ... FAIL
python3.7-pelican> test_custom_generation_works (pelican.tests.test_pelican.TestPelican) ... FAIL
python3.7-pelican> test_cyclic_intersite_links_no_warnings (pelican.tests.test_pelican.TestPelican) ... ok
python3.7-pelican> Test that a warning is issued if MD_EXTENSIONS is used ... ok
(...)
python3.7-pelican> Ran 251 tests in 6.438s
python3.7-pelican> FAILED (SKIP=19, failures=2)
builder for '/nix/store/qjf1b71v9iy12jgv3qcfhlgrchvf9pcy-python3.7-pelican-4.2.0.drv' failed with exit code 1; last 10 log lines:
  pelican.generators: INFO: Copying /build/source/samples/content/extra/robots.txt to robots.txt
  pelican.utils: INFO: Copying /build/source/samples/content/pictures/Fat_Cat.jpg to /build/pelicantests.fanq_r16/pictures/Fat_Cat.jpg
  pelican.generators: INFO: Copying /build/source/samples/content/pictures/Fat_Cat.jpg to pictures/Fat_Cat.jpg
  pelican.utils: INFO: Copying /build/source/samples/content/pictures/Sushi_Macro.jpg to /build/pelicantests.fanq_r16/pictures/Sushi_Macro.jpg
  --------------------- >> end captured logging << ---------------------

  ----------------------------------------------------------------------
  Ran 251 tests in 6.438s

  FAILED (SKIP=19, failures=2)
[0 built (1 failed)]
```

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.3.16, NixOS, 19.09.2260.3b9b10e49be (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.3`
 - channels(root): `"nur, home-manager-19.09, lorri, mozilla-overlay, nixos-19.09.2260.3b9b10e49be, nixos-hardware, nixos-unstable-20.09pre217261.a2e06fc3423, nixpkgs-19.09.2260.3b9b10e49be"`
 - channels(addy): `"home-manager-19.09, lorri, mozilla-overlay, nixos-19.09.2284.bf7c0f0461e, nixos-hardware, nixos-unstable-20.09pre217537.d96bd3394b7, nixpkgs-19.09.2284.bf7c0f0461e, nur"`
 - nixpkgs: `/home/addy/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - python3Packages.pelican
# a list of nixos modules affected by the problem
module:
```

**Comment by Moredread** [comment]
cc @prikhi

**Comment by florianjacob** [comment]
Searching for workarounds on 20.03, I can confrim this is the case for python 3.6, 3.7 and 3.8.

**Comment by florianjacob** [comment]
Could not find out what was actually causing the test to fail. :/ As a workaround, the pelican checks can be disabled, and pelican generally seems to still work for me, so it seems like it is nothing too serious.

```
(python.pkgs.pelican.overrideAttrs (oldAttrs: { doCheck = false; doInstallCheck = false; } ))
```

**Comment by dashmoho** [comment]
I was finally able to trace it to pygments update. E.g. I was not able to build pelican on commit abc938446eb28bbe1f040757589e3192a4c10ec6, but was able with previous one. So we can re-import pygments 2.4 or skip those checks for now and wait for next release of pelican. I see that fix for this issue is already in pelican repository.

**Comment by Moredread** [comment]
Another option would be to fetch that change as a patch for the pelican derivation, if it applies cleanly

**Comment by dashmoho** [comment]
I've extracted patch from https://github.com/getpelican/pelican/commit/9a0a08f460c784053b113a8de709eb4b1bdb6b80. Now I able to build and pass tests without problem. Tested on both master and release-20.03.

**Comment by Moredread** [comment]
A slight improvement would be to get the patch via `fetchpatch` like in https://github.com/NixOS/nixpkgs/blob/df81d9a41c94792e02527dc9fa105153155f2e7c/pkgs/applications/office/scribus/unstable.nix#L46

But already a big thanks for tracking this down. ;)

**Comment by Moredread** [comment]
Closed in favor of PR #87600
